core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -556,7 +569,7 @@ private void registerBuiltinQueryParsers() {
         registerQueryParser(new CommonTermsQueryParser(), CommonTermsQueryParser.QUERY_NAME_FIELD);
         registerQueryParser(new SpanMultiTermQueryParser(), SpanMultiTermQueryParser.QUERY_NAME_FIELD);
         QueryParser<FunctionScoreQueryBuilder> functionScoreParser = (QueryParseContext c) -> FunctionScoreQueryBuilder
-                .fromXContent((String name) -> functionScoreParsers.get(name), c);
+                .fromXContent(this::lookupScoreFunctionParser, c);
```

Comment:
mmm I think I don't see why we need a functional interface here. Seems like having a registry class makes it more clear and is more consistent with what we do with queries. But it's my personal opinion, not sure what others think.